EIPS/eip-5289.md

```diff
@@ -0,0 +1,139 @@
+
+### Signing a Document
+
+Calculate the signature of the following. The [EIP-191](./eip-191) version byte is `0x53`:
+
+```solidity
+keccak256(bytes("\x19Sign Legal Document:") + multihash)
+```
+
+This MUST be compatible with [EIP-1271](./eip-1271.md).
+
+Wallets MUST display a rendering of the linked file to the user.
+
+### Legal Contract Library Interface
+
+```solidity
+/// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+interface IContractLibrary is IERC165 {
+    /// @notice Emitted when signDocument is called
+    event DocumentSigned(address indexed signer, uint48 indexed documentId);
+    
+    /// @notice The IPFS multihash of the legal document. This MUST use a common file format, such as PDF, HTML, TeX, or Markdown.
+    function legalDocument(uint48 documentId) public pure returns (bytes memory);
+    
+    /// @notice Returns whether or not the given user signed the document, and if so, when they did.
+    /// @dev If the user has not signed the document, the timestamp may be anything.
+    function documentSigned(address user, uint48 documentId) public view returns (boolean signed, uint64 timestamp);
+
+    /// @notice Provide a signature
+    /// @dev This MUST be validated by the smart contract. This MUST emit DocumentSigned or throw.
+    function signDocument(address signer, uint48 documentId, bytes memory signature) public;
+}
+```
+
+### Requesting a Signature
+
+To request that certain documents be signed, revert with the following reason:
+
+```solidity
+string.concat("Must Sign Documents:", libraryAddress1, "-", documentId1, ",", libraryAddress2, "-", documentId2, ...)
+```
+
+Example:
+
+```solidity
+"Must Sign Documents:0x0000000000000000000000000000000000000000-1,0x000000000000000000000000000000000000000-2,0x00000000000000000000000000000000000dead-743"
+```
+
+## Rationale
+
+- `uint64` was chosen for the timestamp return type as 64-bit time registers are standard.
+- `uint48` was chosen for the document ID as it uses 26 less bytes than `uint256`, and it is unlikely that more than 4 billion documents will ever need be signed in the same library. If so, oh well, just deploy another library.
+- `signDocument` allows for the signing user to be different than the transaction sender so that it is possible to subsidize this type of transaction.
+- IPFS is mandatory because the authenticity of the signed document can be proven.
+
+## Backwards Compatibility
+
+No backwards compatibility issues found.
+
+## Reference Implementation
+
+### Legal Contract Library
+
+```solidity
+/// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/introspection/ERC165Storage.sol";
+import "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
```

Comment:
```suggestion
import "ERC165Storage.sol";
import "SignatureChecker.sol";
```
It should be an exercise for the reader to decide where to get their dependencies.  That being said, `SignatureChecker` isn't an EIP so this should probably be expanded out at least into an inline interface with comments.